src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4718 @@
+    cXyz local_16c;
+    cXyz cStack_160;
+    float local_ac;
+    float local_a8;
+    float local_a4;
+    
+    cXyz local_ec;
+    cXyz local_e0;
+    cXyz local_dc;
+    cXyz local_d0;
+
+    static cXyz ripple_scale(0.6f, 0.6f, 0.6f);
+    static Vec rope_offset = {160.0f, 0.0f, 0.0f};
+    static Vec water_drop_scale = {1.5f, 1.0f, 1.5f};
+    
+    cXyz* ropeSegments;
+    cXyz* currentRopeSegment;
+    short currentRopeSegmentIndex;
+
+    currentRopeSegmentIndex = this->mCurrentRopeSegmentIndex;
+    
+    if (!currentRopeSegmentIndex) {
+        currentRopeSegment = this->mRopeLine.mpLines->mpSegments;
+        ropeSegments = this->mRopeLineSegments;
+    }
+    else {
+        currentRopeSegment = &this->mRopeLine.mpLines->mpSegments[currentRopeSegmentIndex - 1]; // Couldn't figure out the pointer arithmetic here
+        ropeSegments = this->mRopeLineSegments + currentRopeSegmentIndex;
+    }
+
+    cXyz local_f8 = *currentRopeSegment;
+
+    PSMTXMultVec(this->mpSalvageArmModel->getAnmMtx(1), &rope_offset, currentRopeSegment);
+
+    if (this->mProc == &daShip_c::procCraneUp) {
+        local_ec.set(this->mpLinkModel->getBaseTRMtx()[0][3] - currentRopeSegment->x, 
+                     this->mpLinkModel->getBaseTRMtx()[1][3] - currentRopeSegment->y, 
+                     this->mpLinkModel->getBaseTRMtx()[2][3] - currentRopeSegment->z);
+        
+        local_ec.normalize();
+
+        PSVECScale(&local_ec, &local_ec, 10.0f);
+
+        currentRopeSegment--;
+        ropeSegments--;
+
+        for (int i = this->mCurrentRopeSegmentIndex + -2; i >= 0; i--, currentRopeSegment--, ropeSegments--) {
+            currentRopeSegment->set(*currentRopeSegment + local_ec);
+            ropeSegments->set(cXyz::Zero);
+        }
+    }
+    else {
+        currentRopeSegment--;
+        ropeSegments--;
+
+        for (int i = this->mCurrentRopeSegmentIndex + -2; i >= 0; i--, currentRopeSegment--, ropeSegments--) {
+            local_f8 = *currentRopeSegment;
+
+            if (ropeSegments->y < this->m03F4) {
+                PSVECScale(ropeSegments, ropeSegments, 0.6f);
+            }
+            else {
+                PSVECScale(ropeSegments, ropeSegments, 0.9f);
+            }
+
+            ropeSegments->y -= 3.0f;
+
+            PSVECAdd(currentRopeSegment, ropeSegments, currentRopeSegment);
+
+            local_ec.set(*currentRopeSegment - *(currentRopeSegment + 1));
+
+            fVar17 = local_ec.abs();
+
+            if (fVar17 < 0.01f) {
+                currentRopeSegment->set(*(currentRopeSegment + 1) + l_rope_base_vec);
+            }
+            else {
+                currentRopeSegment->set(*(currentRopeSegment + 1) + (local_ec * 10.0f)/fVar17);
+            }
+
+            cStack_160 = (*currentRopeSegment - local_f8) * 0.05f; // This isn't meant to be storing this result in a local but couldn't figure out how to avoid it given the * 0.05f operation
+
+            PSVECAdd(ropeSegments, &cStack_160, ropeSegments);
```

Comment:
```suggestion
            *ropeSegments += cStack_160;
```
If you ever see a `PSVEC` function being called, it's almost certainly an inline of the cXyz class

Applies in a lot of other places here

Reply:
That's going to be one of the abs inlines. abs2, abs, absXZ, abs2XZ, depending on what the code is doing